CORE-3592: Improve the documentation (#37)
jampp/migratron: docs/running_migration.rst

applied. For this, it has a custom table on the database called ``db_migrations``
which has the path of the executed files and when they were executed.

> ``Migratron`` will work by doing the following:

Reviewer: ```suggestion
``Migratron`` works by doing the following:
```

This isn't a spec, migratron already exists.

- You write a plain SQL file as specified on :ref:`creating_a_migration`

> - It will check it internal database (check `First Time`_),
>   it a migration failed to execute. If that happens,

Reviewer 2: ```suggestion
- It will check in its internal database (check `First Time`_)
  if a migration failed to execute. If that happens,
```

  you need to run the ``cleanup`` subcommand. Check more
  about rollbacking a migration on `What it won't do`_ section

- Check which migrations are missing, and which ones should be
  executed: :term:`PRE`, :term:`POST`, or any. In any case,
  the order of execution of the migrations is based on the filename
  of the migrations

- Execute the missing migrations. For each of these:

  a. Connect to the corresponding database using a command line tool

>      - Hive: beeline
>      G- PrestoSQL/PrestoDB: presto-cli
>      - PostgreSQL: psql

Reviewer 2: ```suggestion
     - Hive: ``beeline``
     - PrestoSQL/PrestoDB: ``presto-cli``
     - PostgreSQL: ``psql``
```

>      and run one of the files. In any case, those command are required

Reviewer 2: ```suggestion
     and run one of the files. In any case, those commands are required
```

     to run the migrations. For more information on how to setup them,
     check the :ref:`Installation instructions <installation>`.

>   b. If the file run sucessfully update the status database. When using Hive

Reviewer 2: ```suggestion
  b. If the file run sucessfully, update the status database. When using Hive
```

>   b. If the file run sucessfully, update the status database. When using Hive

Reviewer: ```suggestion
  b. If the file runs successfully, update the status database. When using Hive
```

     and PrestoSQL/PrestoDB you need to have another PostgreSQL database that

>      it is used to store ``migratron`` internal state. When using PostgreSQL

Reviewer 2: ```suggestion
     is used to store ``migratron`` internal state. When using PostgreSQL
```

>      you might use the same database, or another database.

Reviewer 2: ```suggestion
     you might use the same database, or another one.
```

>      If will also, store the hash of the file, to check if there is some change

Reviewer 2: ```suggestion
     It will also store the hash of the file, to check if there is some change
```

     on the file. It is ok to change the migration file when doing development,

>      but never change the migration file once that it has been executed in

Reviewer 2: ```suggestion
     but never change the migration file once it has been executed in
```

        --db-uri 'jdbc:hive2://localhost:10000/test' \
        --db-type hive \
        --state-db-uri postgres://foo:bar@localhost/test1'

What it won't do
================

This is a list of things that other systems do but ``migratron`` won't do:

- Rollback a migration. If you want to downgrade the last migration or if a

>   migration failed to run, then you must rollback the changes manaully

Reviewer 2: ```suggestion
  migration failed to run, then you must rollback the changes manually.
```

> - The migrations are plain SQL files, so there they might not run on different
>   databases.

Reviewer: ```suggestion
- Adapt migrations to different databases. The migrations are plain SQL files, so they might not run on different
  databases.
```
